server/plugin/plugin.go

```diff
@@ -376,13 +379,58 @@ func (p *Plugin) CreateBotDMPost(userID, message, postType string) {
 	}
 }
 
+func (p *Plugin) CheckIfDuplicateDailySummary(userName, text string) (bool, error) {
+	previousSummary, err := p.GetDailySummaryText(userName)
+	if err != nil {
+		return false, err
+	}
+	if previousSummary == text {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (p *Plugin) StoreDailySummaryText(userName, summaryText string) error {
+	if err := p.API.KVSet(userName+dailySummary, []byte(summaryText)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Plugin) GetDailySummaryText(userName string) (string, error) {
+	summaryByte, err := p.API.KVGet(userName + dailySummary)
+	if err != nil {
+		return "", err
+	}
+
+	return string(summaryByte), nil
+}
+
 func (p *Plugin) PostToDo(info *GitHubUserInfo) {
 	text, err := p.GetToDo(context.Background(), info.GitHubUsername, p.githubConnect(*info.Token))
 	if err != nil {
 		p.API.LogWarn("Failed to get todo text", "userID", info.UserID, "error", err.Error())
 		return
 	}
 
+	if info.Settings.DailyReminderOnChange {
+		isSameSummary, err := p.CheckIfDuplicateDailySummary(info.GitHubUsername, text)
+		if err != nil {
+			p.API.LogWarn("Failed to check the summary on change ", "userID", info.UserID, "error", err.Error())
+			return
+		}
+		if isSameSummary {
+			p.API.LogWarn(" Wont Trigger Daily Summary as the summary is Change ", "NO Summary ", true)
```

Comment:
That seems to spammy to me. Maybe `LogDebug` is fine, but even that seems to much to me.

Reply:
should i remove the message completely , no debug or warn message?

Reply:
Let's just remove the log message here

Reply:
done